src/components/checkbox/GroupableCheckboxReducersUtils.ts

```diff
@@ -0,0 +1,114 @@
+import {IGroupableCheckboxActionPayload} from './GroupableCheckboxActions';
+import {groupableCheckboxInitialState, IGroupableCheckboxesState} from './GroupableCheckboxReducers';
+
+export const addParentCheckbox = (state: IGroupableCheckboxesState,
+                                  action: IReduxAction<IGroupableCheckboxActionPayload>): IGroupableCheckboxesState => {
+  const newCheckbox = {
+    id: action.payload.id,
+    checked: !!action.payload.checked,
+  };
+
+  if (state === groupableCheckboxInitialState) {
+    return {
+      ...state,
+      parentId: action.payload.id,
+      parent: newCheckbox,
+    };
+  }
+
+  return {...state, parent: newCheckbox};
+};
+
+export const addChildCheckbox = (state: IGroupableCheckboxesState,
+                                 action: IReduxAction<IGroupableCheckboxActionPayload>): IGroupableCheckboxesState => {
+  const newChecboxes: ICheckboxState[] = state.checkboxes.concat({
+    id: action.payload.id,
+    checked: !!action.payload.checked,
+  });
+
+  const total: number =  state.total + 1;
+  const nbChecked: number = state.nbChecked + (action.payload.checked ? 1 : 0);
+
+  if (state === groupableCheckboxInitialState) {
+    return {
+      ...state,
+      parentId: action.payload.parentId,
+      total,
+      nbChecked,
+      checkboxes: newChecboxes,
+    };
+  }
+
+  return _.extend({}, state, {
+    checkboxes: newChecboxes,
+    total,
+    nbChecked,
+  });
+};
+
+export const removeParentCheckbox = (state: IGroupableCheckboxesState[],
+                                     action: IReduxAction<IGroupableCheckboxActionPayload>): IGroupableCheckboxesState[] => {
+  return _.reject(state, (checkboxState: IGroupableCheckboxesState) => action.payload.id === checkboxState.parentId);
+};
+
+export const removeChildCheckbox = (state: IGroupableCheckboxesState,
+                                    action: IReduxAction<IGroupableCheckboxActionPayload>) => {
+  const checkboxState: ICheckboxState = _.findWhere(state.checkboxes, {id: action.payload.id});
+  if (checkboxState) {
+    return {
+      checkboxes: _.reject(state.checkboxes, (checkbox: ICheckboxState) => checkbox.id === action.payload.id),
+      nbChecked: state.nbChecked + (checkboxState && checkboxState.checked ? -1 : 0),
+      total: state.total + (checkboxState ? -1 : 0),
+    };
+  }
+  return state;
+};
+
+export const toggleChildCheckbox = (state: IGroupableCheckboxesState,
+                                    action: IReduxAction<IGroupableCheckboxActionPayload>) => {
+  let isChecked: boolean;
+  const checkboxes: ICheckboxState[] = _.map(state.checkboxes, (checkbox: ICheckboxState) => {
+    if (checkbox.id === action.payload.id) {
+      isChecked = !checkbox.checked;
+      return {
+        id: checkbox.id,
+        checked: isChecked,
+      };
+    }
+    return checkbox;
+  });
+  if (checkboxes) {
+    const nbChecked: number = state.nbChecked += (isChecked ? 1 : -1);
```

Comment:
you could probably do `const nbChecked = _.where(state.checkboxes, {checked: true}).length;` to be sure to always have the correct number